### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
# Blog app: add authorization rules
> ***In this feature update:***
- [x] Install CanCanCan in the project.
- [x] Add a `role` column to the 1users` table.
- [x] Using `CanCanCan` for the authorization, a user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`)
- [x] Using `CanCanCan` for the authorization, a user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`)

### ***In collaboration with @Buyaki01***

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, :published => true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
```
Comment (arslanbisharat) [inline app/models/ability.rb]: - [x]  Let's removed the commented code from your project. It makes code hard to read.

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # user ||= User.new # guest user (not logged in)
```
Comment (arslanbisharat) [inline app/models/ability.rb]: - [x]  Remove commented code.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/ability.rb (hunk 1 above)
Comment (Chimwemwe-127001) [reply app/models/ability.rb]: done

File: app/models/ability.rb (hunk 2 above)
Comment (Chimwemwe-127001) [reply app/models/ability.rb]: done

**Review by er-contreras — APPROVED** [review]
Hi @Chimwemwe-127001, 

Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉 

- Well done!

Charlie Brown

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**